packaging/opensuse/snapd.spec

```diff
@@ -143,18 +156,37 @@ updated as soon as new upstream releases are made and is designed to live in
 the system:snappy repository.
 
 %prep
-%setup -q
-
-# Set the version that is compiled into the various executables
+# NOTE: Instead of using setup -q we are unpacking a subdirectory of the source
+# tarball into a directory that is automatically on the future GOPATH. This
+# means that while go doesn't care at all the current working directory is not
+# the top-level directory of the source tarball which some people may find
+# unusual.
+
+# Create indigo compatible build layout.
+mkdir -p %{indigo_srcdir}
+tar -axf %{_sourcedir}/%{name}_%{version}.vendor.tar.xz --strip-components=1 -C %{indigo_srcdir}
+
+# Patch the source in the place it got extracted to.
+pushd %{indigo_srcdir}
```

Comment:
This is fine and in sync with repo packaging https://build.opensuse.org/package/view_file/system:snappy/snapd/snapd.spec?expand=1